pool.go

```diff
@@ -0,0 +1,70 @@
+package Go_Pool
+
+import (
+	"errors"
+	"runtime"
+	"sync"
+)
+
+type Pool struct {
+	Tasks       []*Task
```

Comment:
Since task are submitted via tasksChan, why is this field needed?

Reply:
I limited the tasksChan size to go routines count. To store extra tasks that need to executed whenever the tasksChan will get free will be stored in the Tasks slice.